src/vs/vscode.proposed.d.ts

```diff
@@ -538,4 +538,19 @@ declare module 'vscode' {
 	}
 
 	//#endregion
+
+	//#region Matt: Deinition range
+
+	export interface DefinitionLink {
+		origin?: Range;
+		uri: Uri;
+		range: Range;
+		selectionRange?: Range;
```

Comment:
using these field names make `DefinitionLink` compatible with `Location`

Reply:
sneaky... we could also make `origin` mandatory and use the `position` of the request as `origin`

Reply:
or did you plan that as non-breaking API evolution?

Reply:
Main reason for making it optional is that I think it makes sense for implementers:

- When `origin` is provided: always use this value.
- When `origin` is undefined: use the word range at the request position as the `origin`

If `origin` is not optional, I believe many extensions would have to compute the word range themselves